### DISCUSSION TITLE
How to redirect my request?

### DISCUSSION THREAD

**Discussion by OhhhThatVarun** [go-kit/kit#1162] [Q&A]
So, I want to redirect my request to some other service. But the function ```redirect``` requires access to ```ResponseWriter and *Request```. I'm not getting where to put the redirect. Any help would be appreciated.

**Comment by peterbourgon** [comment]
Why? The answer will inform the best design.

**Reply by Ni3ntnx** [reply]
@peterbourgon  in my application, i have a IAM service which is responsible for performing authentication.  for that i need ability to redirect the request to IAM service's authn endpoint to get authentication done. but as ResponseWriter is not available i am unable to redirect request using https://cs.opensource.google/go/go/+/refs/tags/go1.17.7:src/net/http/server.go;l=2107

**Comment by whiteboy19787bm** [comment]
I would like to try but I would need a tutor

**Comment by xescugc** [comment]
If you want to do it on a "encoder", what I normally use is

```
	w.Header().Set("Location", "/redirected-location")
	w.WriteHeader(http.StatusSeeOther)
```
